alvr/server/src/connection.rs

```diff
@@ -769,10 +769,11 @@ async fn connection_pipeline(
             let (data_sender, mut data_receiver) = tmpsc::unbounded_channel();
             *VIDEO_SENDER.lock() = Some(data_sender);
 
-            while let Some(VideoPacket { header, payload }) = data_receiver.recv().await {
-                let mut buffer = socket_sender.new_buffer(&header, payload.len())?;
-                buffer.get_mut().extend(payload);
-                socket_sender.send_buffer(buffer).await.ok();
+            let mut sender_buffer = SenderBuffer::new();
+            while let Some(VideoPacket { timestamp, payload }) = data_receiver.recv().await {
+                sender_buffer.set_header(&timestamp)?;
+                sender_buffer.payload_mut().extend(payload);
```

Comment:
This concerns me a bit, since this is an additional and possibly large copy. Maybe a channel could already provide such buffers then?

Reply:
We would need to send back the used SendBuffers using another channel. The channel is required at all because we are in async code while the source of the data is in another sync thread. Once we switch to a sync StreamSocket with message-io we can eliminate the copy. I don't think it's worth doing it right now.